SingleResponsibilityPrinciple/Models/Invoice.cs

```diff
@@ -0,0 +1,29 @@
+public class Invoice
```

Comment:
The question here is, am I doing two jobs or one? Is it possible you represent a model, plus send email? Splitting it out is spot on.